pkg/cmd/psa/parser.go

```diff
@@ -0,0 +1,65 @@
+package psa
+
+import (
+	"regexp"
+	"strings"
+
+	"k8s.io/klog/v2"
+)
+
+const headerString = "existing pods in namespace"
+
+var (
+	// wordsInParenthesis parses words from parenthesis.
+	wordsInParenthesis = regexp.MustCompile(`"([^"]+)"`)
+	// submatch in contrast to the whole match.
+	submatch = 1
+	// firstMatch is the first match.
+	firstMatch = 0
+	// secondMatch is the second match.
+	secondMatch = 1
+)
+
+// parseWarnings parses the warnings that are returned by the API request and
+// creates a PodSecurityViolation out of it, that contains the namespace, level,
+// pod name and violations.
+// We expect to have
+//
+// Example Warning Slice:
+// [0] existing pods in namespace "p0t-sekurity-namespace" violate the new PodSecurity enforce level "restricted:latest"
+// [1] p0t-sekurity-pod: allowPrivilegeEscalation != false, unrestricted capabilities, runAsNonRoot != true, seccompProfile
+// OR
+// [1] p0t-sekurity-pod (and 1 other pod): allowPrivilegeEscalation != false, unrestricted capabilities, runAsNonRoot != true, seccompProfile
+// OR
+// [1] p0t-sekurity-pod (and 10 other pods): allowPrivilegeEscalation != false, unrestricted capabilities, runAsNonRoot != true, seccompProfile
+func parseWarnings(warnings []string) *PodSecurityViolation {
+	if len(warnings) == 0 {
+		return nil
+	}
+
+	var psv PodSecurityViolation
+
+	// There should be exactly 2 warnings, but lets not rely on the order.
+	if len(warnings) != 2 {
+		klog.Warningf("Expected 2 warnings, got %d", len(warnings))
+	}
+
+	for _, warning := range warnings {
+		// Namespace Warning Message
+		if strings.HasPrefix(warning, headerString) {
+			// The text should look like "existing pods in namespace "my-namespace" violate the new PodSecurity enforce level "mylevel:v1.2.3"
+			titleMatches := wordsInParenthesis.FindAllStringSubmatch(warning, 2)
+			psv.Namespace = titleMatches[firstMatch][submatch]
+			psv.Level = titleMatches[secondMatch][submatch]
+			continue
+		}
+
+		// The text should look like this: {pod name}{(optional) more pods hint}: {policy warning A}, {policy warning B}, ...
```

Comment:
I could write an admission webhook that creates a warning that says, "I made this error to make krzyz code do wrong things pod: this, isn't right, because, anyone, can make, a webhook".  Can you identify the error you want.